frida_tools/apk.py

```diff
@@ -299,10 +299,11 @@ def append_str(self, add: str) -> int:
         offset = len(chunk_data) - 8 - self.strings_offset + 4
 
         if self.utf8:
-            # UTF-16 length (ignored)
-            chunk_data.extend(struct.pack("<B", len(add)))
-            # UTF-8 length
+            assert len(add.encode("utf-8")) < 128  # multi-byte len strings not supported yet
+            # the length of the string in characters
             chunk_data.extend(struct.pack("<B", len(add)))
+            # the length of the UTF-8 encoding of the string in bytes
+            chunk_data.extend(struct.pack("<B", len(add.encode("utf-8"))))
```

Comment:
```suggestion
            length_in_characters = len(add)
            length_in_bytes = len(add.encode("utf-8"))
            chunk_data.extend(struct.pack("<BB", length_in_characters, length_in_bytes))
```

(Should avoid comments if the code can be expressed such that they're not needed.)